Realm/RLMSyncUtil.mm

```diff
@@ -112,8 +132,81 @@ RLMSyncStopPolicy translateStopPolicy(SyncSessionStopPolicy stop_policy)
     return nullptr;
 }
 
+NSError *make_auth_error_bad_response(NSDictionary *json) {
+    return [NSError errorWithDomain:RLMSyncAuthErrorDomain
+                               code:RLMSyncAuthErrorBadResponse
+                           userInfo:json ? @{kRLMSyncErrorJSONKey: json} : nil];
+}
+
+NSError *make_auth_error_http_status(NSInteger status) {
+    return [NSError errorWithDomain:RLMSyncAuthErrorDomain
+                               code:RLMSyncAuthErrorHTTPStatusCodeError
+                           userInfo:@{kRLMSyncErrorStatusCodeKey: @(status)}];
+}
+
+NSError *make_auth_error_client_issue() {
+    return [NSError errorWithDomain:RLMSyncAuthErrorDomain
+                               code:RLMSyncAuthErrorClientSessionError
+                           userInfo:nil];
+}
+
+NSError *make_auth_error(RLMSyncErrorResponseModel *model) {
+    NSString *description = model.title;
+    return [NSError errorWithDomain:RLMSyncAuthErrorDomain
+                               code:model.code
+                           userInfo:description ? @{NSLocalizedDescriptionKey: description} : nil];
 }
 
+NSError *make_permission_error_get(NSString *description, util::Optional<NSInteger> code) {
+    return make_permission_error(description, std::move(code), true);
+}
+
+NSError *make_permission_error_change(NSString *description, util::Optional<NSInteger> code) {
+    return make_permission_error(description, std::move(code), false);
+}
+
+NSError *make_sync_error(RLMSyncSystemErrorKind kind, NSString *description, NSInteger code, NSDictionary *custom) {
+    NSDictionary *userInfo;
```

Comment:
Yes, there's really nothing that `kind` gets you that passing the underlying error message and string don't do better.